yolov7/utils/sahiModelExtended.py

```diff
@@ -0,0 +1,369 @@
+        Args:
+            shift_amount_list: list of list
+                To shift the box and mask predictions from sliced image to full sized image, should
+                be in the form of List[[shift_x, shift_y],[shift_x, shift_y],...]
+            full_shape_list: list of list
+                Size of the full image after shifting, should be in the form of
+                List[[height, width],[height, width],...]
+        """
+        raise NotImplementedError()
+
+    def _apply_category_remapping(self):
+        """
+        Applies category remapping based on mapping given in self.category_remapping
+        """
+        # confirm self.category_remapping is not None
+        if self.category_remapping is None:
+            raise ValueError("self.category_remapping cannot be None")
+        # remap categories
+        for object_prediction_list in self._object_prediction_list_per_image:
+            for object_prediction in object_prediction_list:
+                old_category_id_str = str(object_prediction.category.id)
+                new_category_id_int = self.category_remapping[old_category_id_str]
+                object_prediction.category.id = new_category_id_int
+
+    def convert_original_predictions(
+        self,
+        shift_amount: Optional[List[int]] = [0, 0],
+        full_shape: Optional[List[int]] = None,
+    ):
+        """
+        Converts original predictions of the detection model to a list of
+        prediction.ObjectPrediction object. Should be called after perform_inference().
+        Args:
+            shift_amount: list
+                To shift the box and mask predictions from sliced image to full sized image, should be in the form of [shift_x, shift_y]
+            full_shape: list
+                Size of the full image after shifting, should be in the form of [height, width]
+        """
+        self._create_object_prediction_list_from_original_predictions(
+            shift_amount_list=shift_amount,
+            full_shape_list=full_shape,
+        )
+        if self.category_remapping:
+            self._apply_category_remapping()
+
+    @property
+    def object_prediction_list(self):
+        return self._object_prediction_list_per_image[0]
+
+    @property
+    def object_prediction_list_per_image(self):
+        return self._object_prediction_list_per_image
+
+    @property
+    def original_predictions(self):
+        return self._original_predictions
+
+
+class Yolov7DetectionModel(DetectionModel):
+    def __init__(
+        self,
+        model_path: Optional[str] = None,
+        model: Optional[Any] = None,
+        config_path: Optional[str] = None,
+        device: Optional[str] = None,
+        mask_threshold: float = 0.5,
+        confidence_threshold: float = 0.25,
+        category_mapping: Optional[Dict] = None,
+        category_remapping: Optional[Dict] = None,
+        load_at_init: bool = True,
+        image_size: int = None,
+        bgr: bool = False,
```

Comment:
is there a point to expose this then if sahi uses PIL? will there be a situation where we need this to be true?

Reply:
For future development maybe; where SAHI no longer user PIL ? cause is still under development